### ISSUE TITLE
VSCode extension setttings are not read correctly

### ISSUE THREAD

**Issue by Godfather95** [microsoft/prompty#49]
I am using the following configuration in settings.json:

```
{
      "name": "default",
      "type": "azure_openai",
      "azure_deployment": "gpt-35-turbo",
      "api_version": "2023-12-01-preview",
      "azure_endpoint": "${env:AZURE_OPENAI_ENDPOINT}",
      "api_key": "${env:AZURE_OPENAI_API_KEY}"
},
{
      "name": "gpt-3.5-turbo",
      "type": "openai",
      "api_key": "${env:OPENAI_API_KEY}",
      "organization": "${env:OPENAI_ORG_ID}",
      "base_url": "${env:OPENAI_BASE_URL}"
}
```

When running a `.prompty` file, I get the following info message in the Prompty Output:

`info] Missing prompty configuration setting: MissingDeployment. Your setting is {"name":"default","type":"azure_openai","api_version":"2023-12-01-preview","azure_endpoint":"https://<EndpointName>.openai.azure.com/","api_key":"<APIKEY>"}`

It seems the Extension doesn't even read the "azure_deployment" from the settings.

**Comment by Godfather95** [comment]
@sethjuarez Yeah, I tried it, and I even copied the settings to be the "non default". But no matter which model configuration I select, I always get the error.

**Comment by anuraj** [comment]
For me as well. I tried changing the configuration, but always getting the error.
![image](https://github.com/user-attachments/assets/55dd116b-477e-4dbd-9d47-dbb82ed27ebf)

**Comment by abakumovoleg** [comment]
the same problem, it always use type azure_openai even if I set openai instead

**Comment by perkops** [comment]
I am experiencing the same issue (even with hardcoded values, not being read from environment).

**Comment by sethjuarez** [comment]
oof - ok - I will look into it - I am adding some features to the extension this week/next. There's a couple of invokers I need to add to make this work

**Comment by AnaTipps** [comment]
I am getting something related, I changed the user settings to add my deployment and after that I start getting this error ![image](https://github.com/user-attachments/assets/d8a51733-468f-451c-881e-f6498b6f3281)
I reverted the changed, uninstall and install again my extension and nothing have worked so far.
I am using the API Key Endpoint pair in my .env file

**Comment by AnaTipps** [comment]
> I am getting something related, I changed the user settings to add my deployment and after that I start getting this error ![image](https://github.com/user-attachments/assets/d8a51733-468f-451c-881e-f6498b6f3281)
> I reverted the changed, uninstall and install again my extension and nothing have worked so far.
> I am using the API Key Endpoint pair in my .env file 

It seems the problem has been resolved. I traced back the changes made and reverted them. The issue appeared to be related to the workspace settings. Once I deleted the .vscode folder that generates when you open the workspace settings file, the problem was resolved.

**Comment by BartNetJS** [comment]
Same issue here

**Comment by MikeYeager** [comment]
Is there a work-around for this?

2024-08-20 12:31:28.296 [info] Missing prompty configuration setting: MissingEndpoint. Your setting is {"name":"default","type":"azure_openai","api_version":"2023-12-01-preview","azure_deployment":"MyChatTest","api_key":"xxxxxxxxxxxxxxxxxxxxxxxxxxx"}

        {
            "name": "default",
            "type": "azure_openai",
            "api_version": "2023-12-01-preview",
            "azure_endpoint": "https://xxxxxxxxxx.openai.azure.com/",
            "azure_deployment": "MyChatTest",
            "api_key": "xxxxxxxxxxxxxxxxxxxxxxxxxxx"
        },

**Comment by wayliums** [comment]
We read the setting from prompty.currentModelConfiguration. 

In some cases VSCode saves to .vscode/settings.json. 
![image](https://github.com/user-attachments/assets/68f10c07-4541-4fd0-85af-a5fe81ec4301)

And some cases it's saved to your user levels settings. 
![image](https://github.com/user-attachments/assets/67b98d56-ef31-48d0-899e-cdc4506d509f)

@BartNetJS @AnaTipps @perkops @abakumovoleg @anuraj @Godfather95 , can you let me know what's your case? I'll see how to make this less problematic

**Comment by MikeYeager** [comment]
In my case, aside from multiple configuration files, the endpoint was overridden in the yaml file itself. I copied it from a sample and didn't notice the setting and didn't remove it and it overrode my settings files. A strategy I've used in the past is to have a way to retrieve all of the current settings to see if they're what is expected. If a value isn't what you expect, it's just a matter of tracking down the source of the unexpected value. Also, a document explaining all the possible places values can be read from and in what precedence order is also useful.

**Comment by MikeStall** [comment]
Same issue here

**Comment by BartNetJS** [comment]
@wayliums In my case I was missing the .env file in the folder of the prompty file
